### DISCUSSION TITLE
dockerizing for production use on AWS/digital ocean/linode etc.

### DISCUSSION THREAD

**Discussion by githubforw0rk** [ialbert/biostar-central#897] [Q&A] [answered]
I noticed there is nothing docker related in this repository.  How difficult would it be to dockerize this repository for use with any of the web hosting providers?  There might be some issues with licenses and Conda as well.

**Comment by ialbert** [comment, marked as answer]
we never built docker images for this purpose thus don't have experience with it

can't be that hard as we use only standard stack, nginx, uwsgi, postgres django

**Reply by githubforw0rk** [reply]
Is there any documentation on how to deploy this app to production?  I would like to start a Q&A site (not related to bioinformatics content) but have been in development locally for a while.  I'm at a loss as to how uwsgi, nginx, postgres, redis and all the other components of this app are hosted.  It doesn't seem like there's much production configuration in this project.  I see a SECRET_KEY which has to be changed in production but am not sure of its purpose.

**Comment by ialbert** [comment]
It is a django app, thus all the rules for DJANGO apply

https://docs.djangoproject.com/en/4.0/howto/deployment/

it is true that, unfortunately, deploying Python apps is nowhere near as simple as deploying PHP. 

Do note that the entire system works out of the box. 

You can just plop waitress in front of it and nothing else needs to be done, everything works out of the box. Select the threaded or even blocking mode for job server.

The Postgres/uwsgi/redis/postgres stack is used to be able to handle dozens of requests per second on a low end machine. The default settings with sqlite ought to work find for sites that only get a few dozen writes per minute. A two-line deploy with waitress should work just fine

https://pypi.org/project/waitress/

**Comment by githubforw0rk** [comment]
I think I see the deployment strategy being used in this project with Ansible and the site_uwsgi.ini configuration.  I believe I just need to change the ansible/makefile/uwsgi configuration to meet my environment's needs with kubernetes/docker/my own host.
